org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLLanguageService.java

```diff
@@ -164,8 +162,7 @@ private static void publishOneDiagnosticInRoot(DOMDocument document, String mess
 	}
 
 	public WorkspaceEdit doRename(DOMDocument xmlDocument, Position position, String newText) {
-		List<TextEdit> textEdits = findDocumentHighlights(xmlDocument, position).stream()
-				.map(h -> new TextEdit(h.getRange(), newText)).collect(Collectors.toList());
+		List<TextEdit> textEdits = XMLRename.getTextEdits(xmlDocument, position, newText);

```

Comment:
Please follow the same instance pattern that other features. In other word XMLRename#getTextEdits should be not static and the service should have a XMLRename rename field.